dbt/adapters/databricks/api_client.py

```diff
@@ -143,91 +38,42 @@ def get_folder(self, _: str, schema: str) -> str:
 
 
 # Switch to this as part of 2.0.0 release
-class UserFolderApi(DatabricksApi, FolderApi):
-    def __init__(self, session: Session, host: str):
-        super().__init__(session, host, "/api/2.0/preview/scim/v2")
+class UserFolderApi(FolderApi):
+    def __init__(self, client: WorkspaceClient):
+        self.client = client
         self._user = ""
 
     def get_folder(self, catalog: str, schema: str) -> str:
         if not self._user:
-            response = self.session.get("/Me")
-
-            if response.status_code != 200:
-                raise DbtRuntimeError(f"Error getting user folder.\n {response.content!r}")
-            self._user = response.json()["userName"]
+            self._user = self.client.current_user.me().user_name or ""
+            if not self._user:
+                raise DbtRuntimeError("Error retrieving current user name")
         folder = f"/Users/{self._user}/dbt_python_models/{catalog}/{schema}/"
         logger.debug(f"Using python model folder '{folder}'")
 
         return folder
 
 
-class WorkspaceApi(DatabricksApi):
-    def __init__(self, session: Session, host: str, folder_api: FolderApi):
-        super().__init__(session, host, "/api/2.0/workspace")
+class WorkspaceApi:
+    def __init__(self, client: WorkspaceClient, folder_api: FolderApi):
+        self.client = client
         self.user_api = folder_api
 
     def create_python_model_dir(self, catalog: str, schema: str) -> str:
         folder = self.user_api.get_folder(catalog, schema)
 
-        # Add
-        response = self.session.post("/mkdirs", json={"path": folder})
-        if response.status_code != 200:
-            raise DbtRuntimeError(
-                f"Error creating work_dir for python notebooks\n {response.content!r}"
-            )
-
+        self.client.workspace.mkdirs(folder)
         return folder
 
     def upload_notebook(self, path: str, compiled_code: str) -> None:
         b64_encoded_content = base64.b64encode(compiled_code.encode()).decode()
-        response = self.session.post(
-            "/import",
-            json={
-                "path": path,
-                "content": b64_encoded_content,
-                "language": "PYTHON",
-                "overwrite": True,
-                "format": "SOURCE",
-            },
+        self.client.workspace.import_(
```

Comment:
Seems like the SDK does not return a response, does it ever throw? https://databricks-sdk-py.readthedocs.io/en/latest/workspace/workspace/workspace.html#databricks.sdk.service.workspace.WorkspaceExt.import_

Reply:
Yes, it will throw, and generally these sorts of exceptions we just allow to bubble up to the user.